mne/beamformer/_dics.py

```diff
@@ -480,52 +1104,86 @@ def tf_dics(epochs, forward, noise_csds, tmin, tmax, tstep, win_lengths,
     Dalal et al. [1]_ used a synthetic aperture magnetometry beamformer (SAM)
     in each time-frequency window instead of DICS.
 
+    An alternative to using noise CSDs is to normalize the forward solution
+    (``normalize_fwd``) or the beamformer weights (``weight_norm``). In
+    this case, ``noise_csds`` may be set to ``None``.
+
     References
     ----------
     .. [1] Dalal et al. Five-dimensional neuroimaging: Localization of the
            time-frequency dynamics of cortical activity.
            NeuroImage (2008) vol. 40 (4) pp. 1686-1700
+    .. [2] Sekihara & Nagarajan. Adaptive spatial filters for electromagnetic
+           brain imaging (2008) Springer Science & Business Media
     """
     _check_reference(epochs)
 
-    if pick_ori not in [None, 'normal']:
-        raise ValueError('Unrecognized orientation option in pick_ori, '
-                         'available choices are None and normal')
-    if len(noise_csds) != len(freq_bins):
-        raise ValueError('One noise CSD object expected per frequency bin')
-    if len(win_lengths) != len(freq_bins):
+    if mode == 'cwt_morlet' and frequencies is None:
+        raise ValueError('In "cwt_morlet" mode, the "frequencies" parameter '
+                         'should be used.')
+    elif mode != 'cwt_morlet' and freq_bins is None:
+        raise ValueError('In "%s" mode, the "freq_bins" parameter should be '
+                         'used.' % mode)
+
+    if frequencies is not None:
+        # Make sure frequencies are always in the form of a list of frequency
+        # bins
+        frequencies = [[f] if isinstance(f, numbers.Number) else f
```

Comment:
I find it more readable if we use a lambda:

```py
convert_to_list = lambda x: [x] if isinstance(x, numbers.Number) else x
frequencies = [convert_to_list(f) for f in frequencies]

Reply:
This will not be PEP8 compliant, PEP8 says you should not define a lambda like this in code (only as parameters to functions)

Reply:
True, but I find it way more readable to suppress this PEP8 directive here, or to use a helper function `def convert_to_list: bla bla` close by, rather than using `[[f] if isinstance(f, numbers.Number) else f for f in frequencies]`